src/cpu.rs

```diff
@@ -0,0 +1,383 @@
+use crate::condition_codes::ConditionCodes;
+use crate::instruction::Instruction;
+use crate::registers::Registers;
+
+#[allow(dead_code)]
+pub struct Cpu {
+    registers: Registers,
+    sp: u8,
+    pub pc: u16,
+    pub memory: [u8; 0xFFFF],
+    condition_codes: ConditionCodes,
+    interrupts_enabled: bool,
+}
+
+impl Cpu {
+    pub fn new() -> Self {
+        Cpu {
+            registers: Registers::new(),
+            sp: 0,
+            pc: 0,
+            memory: [0; 0xFFFF],
+            condition_codes: Default::default(),
+            interrupts_enabled: false,
+        }
+    }
+
+    pub fn execute(&self, instruction: &Instruction) -> (u16, u8) {
+        let pc = match *instruction {
+            Instruction::NOP => self.pc.wrapping_add(instruction.size()),
+            Instruction::JMP(addr) => self.jmp(addr),
+            Instruction::JC(addr) => match self.jc(addr) {
+                None => self.pc.wrapping_add(instruction.size()),
+                Some(next_pc) => next_pc,
+            },
+            Instruction::JNC(addr) => match self.jnc(addr) {
+                None => self.pc.wrapping_add(instruction.size()),
+                Some(next_pc) => next_pc,
+            },
```

Comment:
I've been reading into rust macros and these should be able to be be converted to macros since they are repetitive. Will look into this more later